wiki/decorators.py

```diff
@@ -93,7 +93,7 @@ def wrapper(request, *args, **kwargs):
             #TODO We should try to grab the article form URLPath so the caching is good, and fall back to grabbing it from Article.objects if not
             articles = models.Article.objects
             
-            article = get_object_or_404(articles, id=article_id)
+            article = get_object_or_404(articles, urlpath__slug=article_id)
```

Comment:
There is clearly something wrong here. Looks very suspicious. Well, `article_id` cannot be equal to the `slug` attribute in the child `Slug` model.
Can you explain why this is happening? Perhaps there is something wrong with the names ...

Reply:
Yeah, I found that article_id is not article.id like it was described. It looks like slug. Should I rename it?

Reply:
Thanks for your review and questions. I found that I was wrong and made the decision to do it another way which not affect functionality that already exist.